[ANGLE] Disable CGL and EAGL on Apple Mac Catalyst

emw-apple opened the pull request:
#### 0e6e2460a87d6452655c9d9d6f0bcafa40cc48cc
<pre>
[ANGLE] Disable CGL and EAGL on Apple Mac Catalyst
<a href="https://bugs.webkit.org/show_bug.cgi?id=274558">https://bugs.webkit.org/show_bug.cgi?id=274558</a>
<a href="https://rdar.apple.com/problem/128575633">rdar://problem/128575633</a>

Reviewed by Kimmo Kinnunen and Mike Wyrzykowski.

iOS apps running on an Apple silicon Mac could request the EAGL backend
on older hardware, but now they all use Metal. Stop compiling with EAGL
support. This works towards supporting the building the Mac Catalyst
port using the public SDK, which does not have EAGL.

Also remove CGL (OpenGL) support, as it does not make sense that ASi
would use only Metal but X86_64 would support also CGL.

Currently supported macOS platforms all default to ANGLE/Metal.

* Source/ThirdParty/ANGLE/Configurations/BaseTarget.xcconfig:
* Source/ThirdParty/ANGLE/src/gpu_info_util/SystemInfo_macos.mm:
(angle::GetGpuIDFromDisplayID):
* Source/WebCore/PAL/pal/cocoa/OpenGLSoftLinkCocoa.h:
* Source/WebCore/PAL/pal/cocoa/OpenGLSoftLinkCocoa.mm:
* Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
(WebCore::platformSupportsMetal):
(WebCore::initializeEGLDisplay):
(WebCore::GraphicsContextGLCocoa::platformInitializeContext):
(WebCore::GraphicsContextGLCocoa::platformInitializeExtensions):
(WebCore::needsEAGLOnMac): Deleted.
* Source/WebCore/platform/graphics/cv/GraphicsContextGLCVCocoa.cpp:
(WebCore::GraphicsContextGLCVCocoa::GraphicsContextGLCVCocoa):
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:

Canonical link: <a href="https://commits.webkit.org/279488@main">https://commits.webkit.org/279488@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/5798991d36569d6e54230ebc468cfc523df22723

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53621 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/32979 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/55/builds/6141 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/56921 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/59/builds/4347 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/55923 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/40460 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/61/builds/4158 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/43459 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 wincairo-tests](https://ews-build.webkit.org/#/builders/60/builds/2839 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/55718 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/47/builds/31182 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/46365 "Passed tests") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/41/builds/24593 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/28028 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/64/builds/3682 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/2502 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/49765 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/63/builds/3863 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/58496 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/28783 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/62/builds/3880 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/50857 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| | [  ~~🛠 tv-sim~~](https://ews-build.webkit.org/#/builders/46/builds/29983 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/46525 "Passed tests") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/21/builds/50198 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11689 "Built successfully and passed tests") | [  ~~🛠 watch~~](https://ews-build.webkit.org/#/builders/43/builds/30914 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/29760 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

kkinnunen-apple commented on Source/ThirdParty/ANGLE/Configurations/BaseTarget.xcconfig:
```diff
@@ -32,5 +32,5 @@ USE_HEADERMAP = NO;
 GCC_PREPROCESSOR_DEFINITIONS_macosx = ANGLE_ENABLE_GLSL ANGLE_ENABLE_ESSL ANGLE_ENABLE_OPENGL ANGLE_ENABLE_GL_DESKTOP_BACKEND ANGLE_ENABLE_APPLE_WORKAROUNDS ANGLE_ENABLE_CGL GL_SILENCE_DEPRECATION;
 GCC_PREPROCESSOR_DEFINITIONS_maccatalyst = $(GCC_PREPROCESSOR_DEFINITIONS_macosx);
-GCC_PREPROCESSOR_DEFINITIONS_maccatalyst[arch=arm*] = $(GCC_PREPROCESSOR_DEFINITIONS_macosx) ANGLE_ENABLE_EAGL GLES_SILENCE_DEPRECATION ANGLE_CPU_ARM64;
+GCC_PREPROCESSOR_DEFINITIONS_maccatalyst[arch=arm*] = $(GCC_PREPROCESSOR_DEFINITIONS_macosx) GLES_SILENCE_DEPRECATION ANGLE_CPU_ARM64;
```
remove the GLES_SILENCE_DEPRECATION too

kkinnunen-apple commented on Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:
```diff
@@ -67,9 +67,18 @@ WK_VISIONKITCORE_LDFLAGS_iphonesimulator = -framework VisionKitCore;
-WK_OPENGL_LDFLAGS_cocoatouch = -framework OpenGLES;
+WK_OPENGL_LDFLAGS = $(WK_OPENGL_LDFLAGS_$(WK_PLATFORM_NAME));
+WK_OPENGL_LDFLAGS_macosx = -framework OpenGL;
+WK_OPENGL_LDFLAGS_maccatalyst = -framework OpenGL;
```
the maccatalyst and ios family shouldn't link to opengl anymore

emw-apple replied:
What should we do about `WebGLNoCrashOnOtherThreadAccess`, in TestWebKitAPI? It's using EAGL on iOS and CGL on Catalyst. Seems like we need to remove it or rewrite it to use Metal in order to stop linking TestWebKitAPI against OpenGLES.

And since iOS is building without EAGL, I'm not sure why it's working at all.

emw-apple replied:
In the meantime, I'll have iOS TestWebKitAPI link against GLES to keep the queue green, but if there's an opportunity for easy clean up let's do it.

kkinnunen-apple commented on Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
```diff
@@ -377,14 +360,12 @@ static bool needsEAGLOnMac()
-        if (!enableExtension("GL_EXT_texture_format_BGRA8888"_s))
-            return false;
-    }
+    // For IOSurface-backed textures.
```
to be consistent, this would only be if PLATFORM(MAC)

kkinnunen-apple commented on Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
```diff
@@ -113,13 +113,7 @@ static bool platformSupportsMetal()
-#if PLATFORM(MAC) || PLATFORM(MACCATALYST)
-        // Old Macs, such as MacBookPro11,4 cannot use WebGL via Metal.
-        // This check can be removed once they are no longer supported.
-        return [device supportsFamily:MTLGPUFamilyMac2];
```
Strictly speaking we don't support builds on non Mac2 hw, but this would be still good to preserve for documentation. Also, the mac codebase doesn't make sense otherwise.  So instead, Could you preserve this, and have logic: "no metal" -> "failure". My example amends have an example

emw-apple committed:
[ANGLE] Disable CGL and EAGL on Apple Mac Catalyst
https://bugs.webkit.org/show_bug.cgi?id=274558
rdar://problem/128575633

Reviewed by Kimmo Kinnunen and Mike Wyrzykowski.

iOS apps running on an Apple silicon Mac could request the EAGL backend
on older hardware, but now they all use Metal. Stop compiling with EAGL
support. This works towards supporting the building the Mac Catalyst
port using the public SDK, which does not have EAGL.

Also remove CGL (OpenGL) support, as it does not make sense that ASi
would use only Metal but X86_64 would support also CGL.

Currently supported macOS platforms all default to ANGLE/Metal.

* Source/ThirdParty/ANGLE/Configurations/BaseTarget.xcconfig:
* Source/ThirdParty/ANGLE/src/gpu_info_util/SystemInfo_macos.mm:
(angle::GetGpuIDFromDisplayID):
* Source/WebCore/PAL/pal/cocoa/OpenGLSoftLinkCocoa.h:
* Source/WebCore/PAL/pal/cocoa/OpenGLSoftLinkCocoa.mm:
* Source/WebCore/platform/graphics/cocoa/GraphicsContextGLCocoa.mm:
(WebCore::platformSupportsMetal):
(WebCore::initializeEGLDisplay):
(WebCore::GraphicsContextGLCocoa::platformInitializeContext):
(WebCore::GraphicsContextGLCocoa::platformInitializeExtensions):
(WebCore::needsEAGLOnMac): Deleted.
* Source/WebCore/platform/graphics/cv/GraphicsContextGLCVCocoa.cpp:
(WebCore::GraphicsContextGLCVCocoa::GraphicsContextGLCVCocoa):
* Tools/TestWebKitAPI/Configurations/TestWebKitAPI.xcconfig:

Canonical link: https://commits.webkit.org/279488@main
